### ISSUE TITLE
Error prone interpretation of '.' and ','

### ISSUE THREAD

**Issue by ennob** [ledger/ledger#1941]
In one of my ledgers I noticed an odd behavior in the output of the register report. Suddenly all amounts where multiplied by 1000.

```
19-Feb-04 c3d747cf                    0224a:446e213:5be09a88:c1505dbb        450.000 A        450.000 A
19-Apr-12 8de6fd06                    bbb39:81f3ad1:2c303928:abb1c10e        237.000 A        687.000 A
19-Apr-12 2b3712b1                    e7fbe:e1f903b:a24870ca:f4ef82fc      1,656.200 A      2,343.200 A
19-Aug-28 06e9a695                    4574c:308d2e2:fbba5514:a412b6df        325.000 A      2,668.200 A
19-Sep-03 1107ce13                    9f83d:aa4a64a:47fc59fc:ef9b67b4        200.000 A      2,868.200 A
```

After a lot of poking around I realized that the error was caused by an amount where I had accidentally used . (period) as a thousands separator instead of ',' (comma). Because of that every single number was reinterpreted with both '.' and ',' acting as thousands separators resulting in 450 becoming 450,000 and 1,656.20 becoming 1,656,200.

**Comment by ennob** [comment]
I would much prefer ledger to report an error along the lines of "inconsistent use of separators" rather than silently use the wrong ones. It would also be nice to have configuration variables that can specify the thousands separator and decimal point.

**Comment by tbm** [comment]
fwiw, there is `--decimal-comma`.

**Comment by tbm** [comment]
> After a lot of poking around I realized that the error was caused by an amount where I had accidentally used . (period) as a thousands separator instead of ',' (comma). Because of that every single number was reinterpreted with both '.' and ',' acting as thousands separators resulting in 450 becoming 450,000 and 1,656.20 becoming 1,656,200

I don't think that's what happened or at least I don't see how.  I think what happened is that the numbers of precision increased (i.e. the number of digits after the decimal point).

```
2020-05-01 * Test
    A          1,000.00 USD
    B

2020-05-01 * Test
    A           1.000 USD
    B
```

```
       1,001.000 USD  A
      -1,001.000 USD  B
```

Anyway, do you have a small test case?

**Comment by tbm** [comment]
> 450 becoming 450,000 and 1,656.20 becoming 1,656,200.

In your example 450 became 450.000 (four hundred and fity) and 1,656.20 became 1,656.200.  Which is consistent with what I'm saying.

The problem: how is ledger supposed to know about your typo.  Maybe you wanted extra precision?  People who use crypto use a lot of precision.

**Comment by ennob** [comment]
Yes, I think you're right, the following example supports that explanation

```
2020/06/01 * Small Amount
    Expenses:Stuff                               240 USD
    Assets:Account

2020/07/02 * Incorrect Separator
    Expenses:Stuff                             1.98547473 USD
    Assets:Account
```

The register report of the above is:

```
20-Jun-01 Small Amount          Expenses:Stuff         240.00000000 USD 240.00000000 USD
20-Jul-02 Incorrect Separator   Expenses:Stuff         1.98547473 USD 241.98547473 USD
```

I see the problem of needing more digits of precision but I would expect the full number of digits to be displayed only if I use the `--unround` option. Is there another option I can set to force the output to two digits, or even better to trigger a warning if more than two digits are given in the input file (to help catch errors in numbers copy-pasted from elsewhere).

**Comment by tbm** [comment]
> Is there another option I can set to force the output to two digits, or even better to trigger a warning if more than two digits are given in the input file (to help catch errors in numbers copy-pasted from elsewhere).

I don't think that's possible at the moment.

**Comment by tbm** [comment]
I wonder if this makes sense:

https://hledger.org/journal.html#digit-group-marks
https://hledger.org/journal.html#declaring-commodities

i.e. have a `format` in the `commodity` directive.
